File: Interpreter/StandardFormatter.java
package calculator.xwg;

import android.content.Context;

public class StandardFormatter extends ComplexFormatter {
	StandardFormatter(Context _context){
		super(_context);
	}
	public String toString(Complex complex){
		String valueString = new String();
		if(complex.r != 0){
			if(complex.i != 0){
				valueString += toString(complex.r, 8);
			}else{
				valueString += toString(complex.r, 12);
			}
	    }
		String imaginary;
        if(valueString.length() > 0){
        	imaginary = toString(complex.i, 8);
	    }else{
	    	imaginary = toString(complex.i, 12);
	    }
	    if(imaginary.compareTo("0") != 0){
	    	if(complex.i > 0){
        		valueString += "+";
        	}
	    	imaginary += "i";
            valueString += imaginary;
	    }
	    if(valueString.length() == 0){
	    	valueString = "0";
	    }
	    return valueString;
    }
}

File: Interpreter/StandardFormatter.py
from Interpreter.ComplexFormatter import *


class StandardFormatter(ComplexFormatter):
	def __init__(self, _context):
		ComplexFormatter.__init__(self,_context)

	def toString(self, complex):
		valueString = ''
		if(complex.r != 0):
			if(complex.i != 0):
				valueString = valueString + ComplexFormatter.toStringV(self, complex.r, 8)
			else:
				valueString += valueString + ComplexFormatter.toStringV(self, complex.r, 12)

		imaginary = None
		if(len(valueString) > 0):
			imaginary = ComplexFormatter.toStringV(self, complex.i, 8)
		else:
			imaginary = ComplexFormatter.toStringV(self, complex.i, 12)

		#if(imaginary.compareTo("0") != 0):
		if imaginary != "0":
			if(complex.i > 0):
				valueString = valueString + "+"
				imaginary = imaginary + "i"
				valueString = valueString + imaginary

		if(len(valueString) == 0):
			valueString = "0"

		return valueString
